lib/travis/build/script/r.rb

```diff
@@ -115,6 +124,9 @@ def announce
 
         def install
           super
+          unless setup_cache_has_run_for[:packages]
```

Comment:
I was just following the convention for this I saw in other projects ([c](https://github.com/travis-ci/travis-build/blob/8fb10b052ebf0aa23218931acb0728df4e22978f/lib/travis/build/script/c.rb#L32), [objective_c](https://github.com/travis-ci/travis-build/blob/acca7c1638e46324ec81e6e3efdab7b824b22f31/lib/travis/build/script/objective_c.rb#L44), [python](https://github.com/travis-ci/travis-build/blob/df534f3d3cb79d3dc3761e2259da478f8f1b8f74/lib/travis/build/script/python.rb#L42), [bundler](https://github.com/travis-ci/travis-build/blob/8fb10b052ebf0aa23218931acb0728df4e22978f/lib/travis/build/script/shared/bundler.rb#L25), [cpp](https://github.com/travis-ci/travis-build/blob/8fb10b052ebf0aa23218931acb0728df4e22978f/lib/travis/build/script/cpp.rb#L30)). I guess that convention exists if you want to have multiple caches.

In our case we actually do want to have more than one type of cache in the future, so I actually think it makes sense to keep the hash, but if there is a better way to implement this in ruby I am happy to change it.